app/presenters/publishing_api/featured_documents_presenter.rb

```diff
@@ -0,0 +1,65 @@
+module PublishingApi
+  module FeaturedDocumentsPresenter
+    def featured_documents(featurable_item)
+      featurable_item.feature_list_for_locale(I18n.locale).current.limit(6).map do |feature|
```

Comment:
As far as I can see, we don't specify a limit for featured documents in the Whitehall Admin UI.  Although we do actually cap at 5 (e.g. on [this page](https://www.gov.uk/world/afghanistan/news) which actually has 8 featured documents) 🤯.

There's also a constant specified [here](https://github.com/alphagov/whitehall/blob/27b40a8f9dac2d2f432ad4b2ecf4726bc2ecb480/app/models/featured_link.rb#L4), but that does't appear to be used everywhere (e.g. for organisation pages) 🤯.

Maybe we should add a message to the UI and consider standardising the limit across all of these pages?

Reply:
Ah, a very good point. 6 feels like a sensible limit given the column width (although part of me feels like we should just send them all and let the rendering code decide how many it wants to display). Do you think it would make sense in this PR to leave this as is, add the warning on the UI as you say, and then when we render we can decide whether or not to render 5 (as current) or 6 (as per organisations).

Had a double take on that constant - does not surprise me that there are inconsistently applied limits. But actually I think unless I'm missing something, the constant that you're linking to is actually unrelated to the featured documents - is about the featurable links, not documents, so is applied to organisation pages as far as I can see.

Reply:
It appears we used this constant incorrectly [in the topical event presenter](https://github.com/alphagov/whitehall/blob/27b40a8f9dac2d2f432ad4b2ecf4726bc2ecb480/app/presenters/publishing_api/topical_event_presenter.rb#L64), so that'll probably need fixing.

My preference would be for us to actually limit the number of featured documents that can be added to a page in the first place, so publishers can't add more than the permitted amount and see an error if they do.  That could be done with a simple Rails validation on one of the models.  Then we wouldn't need any limiting at all when presenting the content.  It would also mean publishers won't assume content is appearing as featured, when it actually isn't.

However, I recall we had a discussion about this with topical events and decided to keep the limit at the point of presentation (I don't seem to find the PR or Slack thread though and can't remember the reasoning).

On that basis, I'd suggest keeping the limit in the publishing application so the the value is set as a constant and used in both the logic and message displayed to users.  Otherwise I fear the limit in the publishing and rendering apps could diverge and become different.

Reply:
OK cool, that all makes sense, will have a look at adding constants. Can also raise a pr to fix for topical events.

As a side note, while I was having a look at this I was wondering why topical events had their own special stuff for featured documents. Looks like it was because of the classifications stuff, I [created a card](https://trello.com/c/6duIhmSQ/192-spike-using-shared-featurable-code-for-topical-events-instead-of-custom-topicaleventfeaturingscontroller) to spike just using the same shared featureable stuff